Week3/db-week-3.js

```diff
@@ -0,0 +1,216 @@
+    if (err) {
+      throw err;
+    }
+  }
+});
+
+app.get('/view', async (req, res) => {
+  try {
+    const connection = mysql.createConnection(todoDatabase);
+    const execQuery = util.promisify(connection.query.bind(connection));
+    const listQuery = `select todoid, reminder, item, isdone from todos, items where todos.todoid = items.belongs_to;`;
+    const data = await execQuery(listQuery);
+    res.render('todo', { title: 'Lists', items: data });
+    connection.end();
+  } catch (err) {
+    if (err) {
+      throw err;
+    }
+  }
+});
+
+app.get('/viewOnlyLists', async (req, res) => {
+  try {
+    const connection = mysql.createConnection(todoDatabase);
+    const execQuery = util.promisify(connection.query.bind(connection));
+    const onlyListQuery = `select todoid from todos;`;
+    await execQuery(onlyListQuery).then(data => {
+      if (data.length == 0) {
+        res.render('index', {
+          title: 'Only List',
+          message: "You don't have any lists, please create some.",
+        });
+      } else {
+        res.render('todoList', { title: 'Lists', items: data });
+      }
+    });
+    connection.end();
+  } catch (err) {
+    if (err) {
+      throw err;
+    }
+  }
+});
+
+app.put('/update/:item/:isdone', async (req, res) => {
+  try {
+    const connection = mysql.createConnection(todoDatabase);
+    const execQuery = util.promisify(connection.query.bind(connection));
+    const updateQuery = `UPDATE items SET isdone = '${req.params.item}' Where items.item = '${req.params.isdone}';`;
+    execQuery(updateQuery).then(res.send('Your item is up-to-date!'));
+
+    connection.end();
+  } catch (err) {
+    if (err) {
+      console.log(err);
+    }
+  }
+});
+
+app.get('/delete', async (req, res) => {
+  try {
+    const connection = mysql.createConnection(todoDatabase);
+    const execQuery = util.promisify(connection.query.bind(connection));
+    const showList = `SELECT item, belongs_to from items;`;
+    const lists = await execQuery(showList);
+    res.render('deleteList', {
+      title: 'Delete List',
+      explanation:
+        'So now, for deleting some list items, please use postman. Below, i am leaving you all of your tasks and the lists they belong to',
+      items: lists,
+    });
+  } catch (err) {
+    if (err) {
+      throw err;
+    }
+  }
+});
+
+app.delete('/deleteItem/:item', async (req, res) => {
+  try {
+    const connection = mysql.createConnection(todoDatabase);
+    connection.query(
+      `DELETE FROM items where items.item = ?`,
+      [req.params.item],
+      (err, rows, fields) => {
+        if (!err) res.send('Deleted Succesfuly');
+        else console.log(err);
+      },
+    );
+    connection.end();
+  } catch (err) {
+    if (err) {
+      throw err;
+    }
+  }
+});
+
+app.delete('/deleteList/:list', async (req, res) => {
+  try {
+    const connection = mysql.createConnection(todoDatabase);
+    connection.query(
+      `DELETE FROM todos where todos.todoid = ?`,

```

Comment:
Just be careful since here you're hardcoding the database name `database_name.table_name` but if you change it in the configuration above, it will not work anymore :)